app/src/main/java/com/example/moviedetails/MovieAdapter.kt

```diff
@@ -0,0 +1,36 @@
+package com.example.moviedetails
+
+import android.content.Intent
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.recyclerview.widget.RecyclerView
+import com.bumptech.glide.Glide
+
+class MovieAdapter(private val movies: List<Movie>) :
+    RecyclerView.Adapter<MovieItemRecyclerViewHolder>() {
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MovieItemRecyclerViewHolder {
+        val view =
+            LayoutInflater.from(parent.context).inflate(R.layout.movie_list_item, parent, false)
+        return MovieItemRecyclerViewHolder(view)
+    }
+
+    override fun getItemCount(): Int {
+        return movies.size
+    }
+
+    override fun onBindViewHolder(holder: MovieItemRecyclerViewHolder, position: Int) {
+        holder.movieName.text = movies[position].title
+        holder.releaseYear.text = movies[position].releaseDate
+        val imageUrl = "https://image.tmdb.org/t/p/w200" + movies[position].imageUrl
+        Glide.with(holder.itemView.context).load(imageUrl).into(holder.moviePoster)
+        holder.itemView.setOnClickListener {
+
+            val intent = Intent(it.context, MovieDescriptionActivity::class.java)
+            intent.putExtra(MovieName, holder.movieName.text.toString())
```

Comment:
Seems like you are passing individual properties of Movie data class.
Instead you should pass Movie data class it self. check `Parcelize` feature

Reply:
We should not handle click listener here only. This is makes this Adapter not re-usable in all situations. Move the event up the order and handle it there.